### PR TITLE
Upgrade rules_go, Gazelle, Go SDK, googleapis, com_google_protobuf

### LINKED ISSUE #264: Googleapis is outdated
The current version of googleapis used in this repository is almost 6 years old. This makes importing this repository to other projects unnecessarily cumbersome. Could it be upgraded?

### PR DESCRIPTION
**Pull request by linzhp**
rules_go 0.41.0 no longer comes with `@go_googleapis` repo. This update the references of Google APIs Go code to `@org_golang_google_genproto`. In order to work with the new version of rules_go, this PR also update:

* Gazelle to 0.32.0
* Go SDK to 1.20.6
* com_google_protobuf to 3.15.0
* googleapis to bb964feba5980ed70c9fb8f84fe6e86694df65b0 (Oct 24, 2022)
* org_golang_google_genproto to v0.0.0-20221024183307-1bc688fe9f3e

Fixed #264

This should be merged before #268 and #265 because later versions of googleapis requires latest rules_go

### CODE REVIEW

**Review by EdSchouten — APPROVED** [review]

**Comment by mortenmj** [comment]
@bergsieker do you think we could merge this? I'm interested in having a go at bzlmod support for remote-apis, but it's hindered by the dependencies being out of date.

**Review by werkt — APPROVED** [review]
This seems to simplify and pave the way for bzlmod support. @bergsieker recommend we land this.

**Review by bergsieker — COMMENTED** [review]
I'm concerned that this is actually moving us further from our goal of removing the generated go bindings and making this a proto-only reference repository. However, I acknowledge that we haven't been actively pursuing that goal, so it makes sense not to block progress in order to maintain the status quo.

File: WORKSPACE [hunk 1]
```diff
@@ -18,29 +18,29 @@ bazel_skylib_workspace()
...
+        "https://mirror.bazel.build/github.com/bazelbuild/rules_go/releases/download/v0.41.0/rules_go-v0.41.0.zip",
+        "https://github.com/bazelbuild/rules_go/releases/download/v0.41.0/rules_go-v0.41.0.zip",
     ],
 )
 
 # Gazelle, which we need in order to selectively pull in Gazelle dependencies for Go.
 http_archive(
     name = "bazel_gazelle",
-    sha256 = "62ca106be173579c0a167deb23358fdfe71ffa1e4cfdddf5582af26520f1c66f",
+    sha256 = "29218f8e0cebe583643cbf93cae6f971be8a2484cdcfa1e45057658df8d54002",
```
[the remark this answers is not in the document]
Comment (linzhp) [reply WORKSPACE]: good catch. Updated it

**Review by bergsieker — APPROVED** [review]

**Pull request merged**